### PR TITLE
Add CSS selector filter

### LINKED ISSUE #273: Add support for a HTML filter using CSS selectors
Split off from #226, it was suggested that there should be a filter option for using CSS selectors to pick something from the document.

We probably can't use anything from the standard Python library, but something like this should work:
https://cssselect.readthedocs.io/en/latest/

Note that with Requests-HTML (which we already have as an optional dependency for the "browser" job) it would also work: https://html.python-requests.org

### PR DESCRIPTION
**Pull request by cfbao**
Close #273

### CODE REVIEW

**Comment by cfbao** [comment]
Implement class `LxmlParser` which does all the real work for `CssFilter` and `XPathFilter`.
Similar to the relation and interface between `ElementsBy` and `GetElementBy*`.

File: README.md [hunk 1]
```diff
@@ -150,6 +151,17 @@ filter: xpath:/body
...
+To filter based on a [CSS selector](https://www.w3.org/TR/2011/REC-css3-selectors-20110929/),
+you can use the `css` filter like so:
+
+```yaml
+url: https://example.net/
+filter: css:body
+```
+
+Some limitatons and extensions exist as explained in
+[cssselect's documentatons](https://cssselect.readthedocs.io/en/latest/#supported-selectors).
```
Comment (thp) [inline README.md]: documentatons -> documentation

File: README.md [hunk 2]
```diff
@@ -150,6 +151,17 @@ filter: xpath:/body
...
 
+To filter based on a [CSS selector](https://www.w3.org/TR/2011/REC-css3-selectors-20110929/),
+you can use the `css` filter like so:
+
+```yaml
+url: https://example.net/
+filter: css:body
+```
+
+Some limitatons and extensions exist as explained in
```
Comment (thp) [inline README.md]: limitatons -> limitations

**Comment by cfbao** [comment]
Oops! Fixed.

**Comment by thp** [comment]
Thanks!

**Pull request merged**